src/main/java/de/tum/in/www1/artemis/config/migration/entries/BambooMigrationService.java

```diff
@@ -0,0 +1,499 @@
+
+    public BambooMigrationService(@Qualifier("bambooRestTemplate") RestTemplate restTemplate, BambooService bambooService, BambooInternalUrlService bambooInternalUrlService) {
+        this.restTemplate = restTemplate;
+        this.bambooService = bambooService;
+        this.bambooInternalUrlService = bambooInternalUrlService;
+    }
+
+    /**
+     * Returns a list of all data-item-ids in the given html.
+     *
+     * @param html response that contains a list of items with the data-item-id attribute
+     * @return a list of all data-item-ids in the given html
+     */
+    private static List<Long> getDataItemIds(String html) {
+        if (html == null) {
+            return List.of();
+        }
+        Elements repositories = Jsoup.parse(html).select(".item");
+        List<Long> ids = new ArrayList<>();
+        for (Element repository : repositories) {
+            String repositoryIdString = repository.attr("data-item-id");
+            ids.add(Long.parseLong(repositoryIdString));
+        }
+        return ids;
+    }
+
+    /**
+     * Returns the name of the repository with the given id.
+     *
+     * @param html Response to the request to the editChainRepository.action endpoint
+     * @param id   The id of the repository we need the name for
+     * @return the name of the repository with the given id
+     */
+    private static Optional<String> getRepositoryNameById(String html, Long id) {
+        if (html == null) {
+            return Optional.empty();
+        }
+        Elements repositories = Jsoup.parse(html).select(".item");
+        for (Element repository : repositories) {
+            var link = Jsoup.parse(repository.html()).selectFirst("a");
+            if (link == null) {
+                continue;
+            }
+            String repositoryLink = link.attr("href");
+            Element nameElement = repository.selectFirst("h3");
+            if (repositoryLink.contains("repositoryId=" + id.toString()) && nameElement != null) {
+                return Optional.of(nameElement.text());
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void overrideBuildPlanNotification(String projectKey, String buildPlanKey, VcsRepositoryUrl vcsRepositoryUrl) {
+        List<Long> notificationIds = getAllArtemisBuildPlanServerNotificationIds(buildPlanKey);
+
+        for (var id : notificationIds) {
+            deleteBuildPlanServerNotificationId(buildPlanKey, id);
+        }
+
+        createBuildPlanServerNotification(buildPlanKey, artemisServerUrl + NEW_RESULT_RESOURCE_API_PATH);
+    }
+
+    @Override
+    public void deleteBuildTriggers(String projectKey, String buildPlanKey, VcsRepositoryUrl repositoryUrl) {
+        List<Long> triggerIds = getAllTriggerIds(buildPlanKey);
+        if (triggerIds.size() != 1) {
+            log.warn("The build plan {} has {} triggers", buildPlanKey, triggerIds.size());
```

Comment:
It isn't a problem; it simply highlights something out of the ordinary. Typically, a build plan should have only one trigger. If it has more, we log it but remove all because the new workflow does not need any triggers in Bamboo.